### ISSUE TITLE
The score returned by qnamaker  is not dealed properly in activeLearningUtils

### ISSUE THREAD

**Issue by xiaolangxyz** [microsoft/botbuilder-js#2747]
## Versions
botbuilder-ai": "^4.10.1",

## Describe the bug
The score is not dealed properly in activeLearningUtils.

Qnamaker returns the answers' score between 0 and 100.
But here the score is supposed to be between 0 and 1.

```typescript

public static getLowScoreVariation(qnaSearchResults: QnAMakerResult[]): QnAMakerResult[] {
        if (qnaSearchResults == null || qnaSearchResults.length == 0) {
            return [];
        }

        if (qnaSearchResults.length == 1) {
            return qnaSearchResults;
        }

        let filteredQnaSearchResult = [];
        let topAnswerScore = qnaSearchResults[0].score * 100;

        if (topAnswerScore > ActiveLearningUtils.MaximumScoreForLowScoreVariation) {
            filteredQnaSearchResult.push(qnaSearchResults[0]);
            return filteredQnaSearchResult;
        }

        let prevScore = topAnswerScore;

        if (topAnswerScore > ActiveLearningUtils.MinimumScoreForLowScoreVariation) {
            filteredQnaSearchResult.push(qnaSearchResults[0]);

            for (let i = 1; i < qnaSearchResults.length; i++) {
                if (ActiveLearningUtils.includeForClustering(prevScore, qnaSearchResults[i].score * 100, PreviousLowScoreVariationMultiplier) && this.includeForClustering(topAnswerScore, qnaSearchResults[i].score * 100, MaxLowScoreVariationMultiplier)) {
                    prevScore = qnaSearchResults[i].score * 100;
                    filteredQnaSearchResult.push(qnaSearchResults[i]);
                }
            }
        }

        return filteredQnaSearchResult;
    }

```

**Comment by Zerryth** [comment]
@xiaolangxyz You're right that the Generate Answer API from QnA Maker service directly returns scores in range of 0 - 100 and that the static `getLowScoreVariation` method requires that the score be between 0 - 1, however this is intentional, although it's not clearly obvious why without stepping through the SDK. 🤨 
___
**Generate Answer API**
In the SDK, when you call the [`QnAMaker.getAnswers`](https://github.com/microsoft/botbuilder-js/blob/master/libraries/botbuilder-ai/src/qnaMaker.ts#L133) method, it:
1. Calls the Generate Answer API
2. Gets the result
3. And then [formats the result](https://github.com/microsoft/botbuilder-js/blob/master/libraries/botbuilder-ai/src/qnamaker-utils/generateAnswerUtils.ts#L140)
    - This includes converting the 0 - 100 ranged answer to 0 - 1 range

Why this 3rd step happens, I'm not sure to be honest, but it's been that way in the SDK for years now. 🤷 

___
**QnAMakerDialog and Active Learning**
If you use sample [49.qnamaker-all-features](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/javascript_nodejs/49.qnamaker-all-features), it includes using Active Learning, which is embedded within the [QnAMakerDialog](https://github.com/microsoft/botbuilder-js/blob/master/libraries/botbuilder-ai/src/qnaMakerDialog.ts#L258). Within this class it:
1. Calls Generate Answer API
2. Calls ActiveLearningUtils.getLowScoreVariation, to verify if there are multiple answers that returned from the QnA service that are close in score to each other

Because the Generate Answer API *in the SDK* formats the `QnAResult` to 0 - 1 range, when it calls `ActiveLearningUtils.getLowScoreVariation`, it is in the appropriate range for it to process properly. 

___

So in summary, Active Learning in the SDK intentionally handles score in range of 0 - 1, which is how Generate Answer API in the SDK formats its *QnAResult*s. Actually the whole QnAMaker library handles scores in range of 0 - 1, which you can see if you look at `QnAMakerDialog`, for example. We can't make the change to handle scores from 0 - 100, because that would be a breaking behavioral change in the SDK, unfortunately. If you want to use `getLowScoreVariation` directly, simply have it in your business logic to convert your results' scores to 0 - 1 range before passing them into the method.
